components/access_history_modal/index.ts

```diff
@@ -2,20 +2,22 @@
 // See LICENSE.txt for license information.
 
 import {connect} from 'react-redux';
-import {bindActionCreators} from 'redux';
+import {bindActionCreators, Dispatch} from 'redux';
 import {getUserAudits} from 'mattermost-redux/actions/users';
 import {getCurrentUserId, getUserAudits as getCurrentUserAudits} from 'mattermost-redux/selectors/entities/users';
+import {GlobalState} from 'mattermost-redux/types/store';
+import {GenericAction} from 'mattermost-redux/types/actions';
 
-import AccessHistoryModal from './access_history_modal.jsx';
+import AccessHistoryModal from './access_history_modal';
 
-function mapStateToProps(state) {
+function mapStateToProps(state: GlobalState) {
     return {
         currentUserId: getCurrentUserId(state),
         userAudits: getCurrentUserAudits(state) || [],
     };
 }
 
-function mapDispatchToProps(dispatch) {
+function mapDispatchToProps(dispatch: Dispatch<GenericAction>) {
```

Comment:
just FYI, we have DispatchFunc type defined for this

Reply:
DispatchFunc doesn't pass tsc. I thought that was what we were supposed to use as well.

Reply:
`DispatchFunc` includes stuff for redux-thunk which wouldn't be directly compatible with the types from Redux itself. redux-thunk might provide better types for it, but I haven't looked into it

Reply:
@reflog any thoughts?

Reply:
@hmhealey - I think you are talking about a different func definition.
We have our own version that should work:
@devinbinnie 
`import {DispatchFunc} from 'types/actions'`

Reply:
@reflog It doesn't:
```
components/access_history_modal/index.ts:28:41 - error TS2769: No overload matches this call.
  The last overload gave the following error.
    Argument of type '(dispatch: DispatchFunc) => { actions: any; }' is not assignable to parameter of type 'MapDispatchToPropsParam<{ actions: any; }, {}>'.
      Type '(dispatch: DispatchFunc) => { actions: any; }' is not assignable to type 'MapDispatchToPropsFactory<{ actions: any; }, {}>'.
        Types of parameters 'dispatch' and 'dispatch' are incompatible.
          Types of parameters 'action' and 'action' are incompatible.
            Type 'Action' is not assignable to type 'Action<any>'.
              Property 'type' is missing in type 'ActionFunc' but required in type 'Action<any>'.

28 export default connect(mapStateToProps, mapDispatchToProps)(AccessHistoryModal);
```

Reply:
@devinbinnie - I've checked - you are correct. I'm going to open an issue for myself to fix this Redux incompatibility.

Reply:
Checked with @reflog and he confirmed that there's no clean type mapping at the present, so we'll go with my current solution.